src/os/bluestore/BlueStore.cc

```diff
@@ -5696,6 +5721,21 @@ int BlueStore::_open_super_meta()
     }
   }
 
+  // Min min_alloc_size
+  {
+    bufferlist bl;
+    min_min_alloc_size = 0;
```

Comment:
It is always set after call to _save_min_min_alloc_size. Taking value 0 as argument to this function identifies the condition where we always need to save the min_min_alloc_size. 
The min_alloc_size is set before call to this function. So the cases:

1. min_min_alloc_size : 0, then we need to save current min_alloc_size as min_min_alloc_size. That handles upgrade and mkfs cases where there is no older value.
2. min_min_alloc_size < min_alloc_size, we need to put message and do nothing in this case.
3. min_min_alloc_size > min_alloc_size, we need to put message and save new value.

In all cases at end of this function, min_min_alloc_size == min_alloc_size.

I think it handles all cases ?